pkg/operator/controller/controller.go

```diff
@@ -90,15 +91,43 @@ func New(mgr manager.Manager, config operatorconfig.Config) (controller.Controll
 		return nil, err
 	}
 
-	caBundleCMToDNS := func(o client.Object) []reconcile.Request {
+	objectToDNS := func(o client.Object) []reconcile.Request {
 		return []reconcile.Request{{DefaultDNSNamespaceName()}}
 	}
 	isInNS := func(namespace string) func(o client.Object) bool {
 		return func(o client.Object) bool {
 			return o.GetNamespace() == namespace
 		}
 	}
-	if err := c.Watch(&source.Kind{Type: &corev1.ConfigMap{}}, handler.EnqueueRequestsFromMapFunc(caBundleCMToDNS), predicate.NewPredicateFuncs(isInNS(GlobalUserSpecifiedConfigNamespace))); err != nil {
+	if err := c.Watch(&source.Kind{Type: &corev1.ConfigMap{}}, handler.EnqueueRequestsFromMapFunc(objectToDNS), predicate.NewPredicateFuncs(isInNS(GlobalUserSpecifiedConfigNamespace))); err != nil {
+		return nil, err
+	}
+	// If a node is created or deleted, then the controller may need to
+	// reconcile the DNS service in order to add or remove the
+	// service.kubernetes.io/topology-aware-hints annotation, but only if
+	// the node isn't ignored for the purpose of determining whether to
+	// enable topology-aware hints.
+	nodePredicate := func(o client.Object) bool {
+		node := o.(*corev1.Node)
+		return !ignoreNodeForTopologyAwareHints(node)
+	}
+	if err := c.Watch(&source.Kind{Type: &corev1.Node{}}, handler.EnqueueRequestsFromMapFunc(objectToDNS), predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool { return nodePredicate(e.Object) },
+		DeleteFunc: func(e event.DeleteEvent) bool { return nodePredicate(e.Object) },
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			old := e.ObjectOld.(*corev1.Node)
+			new := e.ObjectOld.(*corev1.Node)
```

Comment:
These are the same objects no?

Reply:
Yeah.  Thanks!  Fixed in https://github.com/openshift/cluster-dns-operator/compare/781c2b876f81169e3461eaf7b724de68a7759ada..55550c70282fc27d55b99dddcb0b8ba49b73dcdc.